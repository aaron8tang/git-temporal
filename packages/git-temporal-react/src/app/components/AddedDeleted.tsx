import React from 'react';
import { style } from 'app/styles';

import { CommaNumber } from 'app/components/CommaNumber';

export interface AddedDeletedProps {
  linesAdded: number;
  linesDeleted: number;
  style?: object;
}

// numbers for humans have commas
export const AddedDeleted = (props: AddedDeletedProps): JSX.Element => {
  return (
    <span style={style('smallerText', { float: 'right' }, props.style)}>
      <span style={style('linesAdded')}>
        +<CommaNumber value={props.linesAdded} />
      </span>
      <span> / </span>
      <span style={style('linesDeleted')}>
        -<CommaNumber value={props.linesDeleted} />
      </span>
    </span>
  );
};
