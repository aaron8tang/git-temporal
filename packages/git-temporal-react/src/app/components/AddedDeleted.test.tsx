import React from 'react';
import { shallow } from 'enzyme';

import { AddedDeleted } from './AddedDeleted';

describe('components/AddedDeleted', () => {
  describe('when rendered epoch time 1539369263 (Fri, 12 Oct 2018 18:34:23 GMT)', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <AddedDeleted linesAdded={20034} linesDeleted={9877349} />
      );
    });

    test('it should match snapshot (it should have rendered linesAdded and linesDeleted with commas)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
