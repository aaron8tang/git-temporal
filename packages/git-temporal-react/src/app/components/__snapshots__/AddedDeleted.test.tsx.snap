// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AddedDeleted when rendered epoch time 1539369263 (Fri, 12 Oct 2018 18:34:23 GMT) it should match snapshot (it should have rendered linesAdded and linesDeleted with commas) 1`] = `
<span
  style={
    Object {
      "color": "#333333",
      "float": "right",
      "fontSize": 11,
      "fontWeight": "normal",
    }
  }
>
  <span
    style={
      Object {
        "color": "green",
      }
    }
  >
    +
    <Component
      value={20034}
    />
  </span>
  <span>
     / 
  </span>
  <span
    style={
      Object {
        "color": "red",
      }
    }
  >
    -
    <Component
      value={9877349}
    />
  </span>
</span>
`;
